pkg/minikube/bootstrapper/kicbs/kicbs.go

```diff
@@ -122,28 +130,286 @@ func (k *Bootstrapper) UpdateCluster(cfg config.MachineConfig) error {
+	}
+
+	if err := verify.SystemPods(c, start, "127.0.0.1", int(k8s.HostBindPort), timeout); err != nil {
+		return errors.Wrap(err, "wait for system pods")
+	}
+
+	return nil
 }
 
 func (k *Bootstrapper) DeleteCluster(config.KubernetesConfig) error {
 	return fmt.Errorf("the DeleteCluster is not implemented in kicbs yet")
 }
-func (k *Bootstrapper) WaitForCluster(config.KubernetesConfig, time.Duration) error {
-	return fmt.Errorf("the WaitForCluster is not implemented in kicbs yet")
-}
+
 func (k *Bootstrapper) LogCommands(bootstrapper.LogOptions) map[string]string {
 	return map[string]string{}
 }
-func (k *Bootstrapper) SetupCerts(cfg config.KubernetesConfig) error {
-	return fmt.Errorf("the SetupCerts is not implemented in kicbs yet")
-}
+
 func (k *Bootstrapper) GetKubeletStatus() (string, error) {
 	return "", fmt.Errorf("the GetKubeletStatus is not implemented in kicbs yet")
 }
 func (k *Bootstrapper) GetAPIServerStatus(net.IP, int) (string, error) {
 	return "", fmt.Errorf("the GetAPIServerStatus is not implemented in kicbs yet")
 }
+
+// client sets and returns a Kubernetes client to use to speak to a kubeadm launched apiserver
+func (k *Bootstrapper) client(k8s config.KubernetesConfig) (*kubernetes.Clientset, error) {
+	if k.k8sClient != nil {
+		return k.k8sClient, nil
+	}
+
+	config, err := kapi.ClientConfig(k.contextName)
+	if err != nil {
+		return nil, errors.Wrap(err, "client config")
+	}
+
+	endpoint := fmt.Sprintf("https://%s", net.JoinHostPort("127.0.0.1", fmt.Sprint(k8s.HostBindPort)))
+	if config.Host != endpoint {
+		glog.Errorf("Overriding stale ClientConfig host %s with %s", config.Host, endpoint)
+		config.Host = endpoint
+	}
+	c, err := kubernetes.NewForConfig(config)
+	if err == nil {
+		k.k8sClient = c
+	}
+	return c, err
+}
+
+// adjustResourceLimits makes fine adjustments to pod resources that aren't possible via kubeadm config.
+func (k *Bootstrapper) adjustResourceLimits() error {
+	rr, err := k.c.RunCmd(exec.Command("/bin/bash", "-c", "cat /proc/$(pgrep kube-apiserver)/oom_adj"))
+	if err != nil {
+		return errors.Wrapf(err, "oom_adj check cmd %s. ", rr.Command())
+	}
+	glog.Infof("apiserver oom_adj: %s", rr.Stdout.String())
+	// oom_adj is already a negative number
+	if strings.HasPrefix(rr.Stdout.String(), "-") {
+		return nil
+	}
+	glog.Infof("adjusting apiserver oom_adj to -10")
+
+	// Prevent the apiserver from OOM'ing before other pods, as it is our gateway into the cluster.
+	// It'd be preferable to do this via Kubernetes, but kubeadm doesn't have a way to set pod QoS.
+	if _, err = k.c.RunCmd(exec.Command("/bin/bash", "-c", "echo -10 | sudo tee /proc/$(pgrep kube-apiserver)/oom_adj")); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("oom_adj adjust"))
+	}
+
+	return nil
+}
+
+// applyOverlayNetwork applies the CNI plugin needed to make kic work
+func (k *Bootstrapper) applyOverlayNetwork() error {
+	cmd := exec.Command(
+		"kubectl", "create", "--kubeconfig=/etc/kubernetes/admin.conf",
```

Comment:
It's OK for this iteration - but we should find a way to make this work without kubectl. Otherwise, we depend on the user having the correct version installed properly.